### PR TITLE
Add Thrift 0.17 module

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
Thrift 0.17 has come out.

Modifications:
- Add Thrift 0.17 module
- Restructure Thrift modules under `thrift` directory.

Result:
- You can use Thrift 0.17 with Armeria.

### CODE REVIEW

File: gradle/scripts/lib/java-publish.gradle [hunk 1]
```diff
@@ -129,11 +129,8 @@ configure(projectsWithFlags('publish')) {
...
-    // A version catalog need to publish Gradle metadata together to be correctly imported externally.
-    if (!hasFlags('version-catalog')) {
-        // For now, disable Gradle metadata generation until verifying it is correctly generated.
-        tasks.withType(GenerateModuleMetadata) {
-            enabled = false
-        }
+    // For now, disable Gradle metadata generation until verifying it is correctly generated.
+    tasks.withType(GenerateModuleMetadata) {
+        enabled = false
     }
```
Comment (ikhoon) [inline gradle/scripts/lib/java-publish.gradle:135]: Oops, it should not be reverted. Maybe I missed committing the change to `gradle-scripts`. Let me update `gradle-scripts` first.

File: .github/workflows/actions_build.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        on: [ self-hosted, macos-latest, windows-latest ]
+        on: [ self-hosted, macos-12, windows-latest ]
```
[the remark this answers is not in the document]
Comment (minwoox) [reply .github/workflows/actions_build.yml:28]: `macos-latest` is `macos-11` and I thought it was okay to use the latest version. 
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

File: gradle/scripts/lib/java-publish.gradle (hunk 1 above)
Comment (minwoox) [reply gradle/scripts/lib/java-publish.gradle]: Updated. 😉

File: .github/workflows/actions_build.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ikhoon) [reply .github/workflows/actions_build.yml:28]: I didn't know `macos-latest` indicates the legacy version.

File: thrift/thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/text/StructContext.java [hunk 3]
```diff
@@ -162,14 +162,15 @@ private static Class<?> getCurrentThriftMessageClass() {
      * Compute a new field name map for the current thrift message
      * we are parsing.
      */
-    private Map<String, TField> computeFieldNameMap(Class<?> clazz) {
+    private <T extends TBase<T, F>, F extends TFieldIdEnum> Map<String, TField> computeFieldNameMap(
```
[the remark this answers is not in the document]
Comment (ikhoon) [reply thrift/thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/text/StructContext.java:165]: I see. Thanks for the detailed explanation.

**Review by ikhoon — APPROVED** [review]
Thanks for reorganizing the thrift modules. 🙇‍♂️

Could you resolve the conflicts?

**Review by jrhee17 — APPROVED** [review]
Looks good! Thanks @minwoox ! 🙇 👍 🙇

**Pull request merged**